### ISSUE TITLE
ModelAuthTransformer assumes all queries and mutations are defined

### ISSUE THREAD

**Issue by onlybakam** [aws-amplify/amplify-cli#55]
**Do you want to request a feature or report a bug?**

bug

**What is the current behavior?**

```
TypeError: Cannot read property 'Properties' of undefined, at ModelAuthTransformer.protectUpdateOrDeleteMutation (xxxxxxx/amplify-cli/packages/graphql-auth-transformer/src/ModelAuthTransformer.ts:327:59)
```

**Steps to reproduce the issue:**

1. create a model that only defines some mutations/queries

```graphql
type Post
  @model(mutations: { create: "createPost" })
  @auth(rules: [{ allow: owner }]) {
}
```

2. amplify api gql-compile

**What is the expected behavior?**

should not return an error. if a resolver does not exist, should not try to modify it. simply skip it.
following could be added/modified in the "protect" functions:

```typescript
        const resolver = ctx.getResource(resolverResourceId)
        if (!rules || rules.length === 0 || !resolver) {
            return
        }
```

**Additional environment details**

* **OS version (ie Windows 10 build X, macOS Sierra 10.12.6, etc.)** mac
* **Output of ``amplify --version``**: 0.1.13
* **Did this work in previous versions?**: N

**Comment by mikeparisstuff** [comment]
Good find. I'll get this fixed.

**Comment by mikeparisstuff** [comment]
Thanks for surfacing this issue. I have fixed it in the above PR.

**Comment by kaustavghosh06** [comment]
Fixed pushed with latest npm publish -> 0.1.18
